benches/wallet_state.rs

```diff
@@ -0,0 +1,204 @@
+use std::env;
+use std::path::Path;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use divan::Bencher;
+use neptune_cash::api::export::KeyType;
+use neptune_cash::api::export::NativeCurrencyAmount;
+use neptune_cash::api::export::NeptuneProof;
+use neptune_cash::api::export::Network;
+use neptune_cash::api::export::ReceivingAddress;
+use neptune_cash::api::export::Timestamp;
+use neptune_cash::api::export::Transaction;
+use neptune_cash::api::export::TransactionDetails;
+use neptune_cash::api::export::TransactionProof;
+use neptune_cash::api::export::TxInput;
+use neptune_cash::config_models::cli_args;
+use neptune_cash::config_models::data_directory::DataDirectory;
+use neptune_cash::models::blockchain::block::validity::block_primitive_witness::BlockPrimitiveWitness;
+use neptune_cash::models::blockchain::block::validity::block_proof_witness::BlockProofWitness;
+use neptune_cash::models::blockchain::block::Block;
+use neptune_cash::models::blockchain::block::BlockProof;
+use neptune_cash::models::blockchain::transaction::primitive_witness::PrimitiveWitness;
+use neptune_cash::models::state::wallet::transaction_output::TxOutput;
+use neptune_cash::models::state::wallet::wallet_configuration::WalletConfiguration;
+use neptune_cash::models::state::wallet::wallet_entropy::WalletEntropy;
+use neptune_cash::models::state::wallet::wallet_state::WalletState;
+use num_traits::CheckedSub;
+use rand::distr::Alphanumeric;
+use rand::distr::SampleString;
+use rand::Rng;
+
+fn main() {
+    divan::main();
+}
+
+mod maintain_membership_proofs {
+    use super::*;
+
+    /// Maintain 100 membership proofs, while receiving an additional 100 UTXOs.
+    mod maintain_100_100 {
+
+        use super::*;
+        use crate::helper::*;
+
+        fn update_wallet_with_block2(bencher: Bencher) {
+            let rt = tokio::runtime::Runtime::new().unwrap();
+            let mut wallet_state = rt.block_on(devops_wallet_state_genesis(Network::Main));
```

Comment:
style: consider a single block_on() with a single async block.

(except for the bench closure which needs its own)